src/Microsoft.PowerShell.ConsoleHost/host/msh/ManagedEntrance.cs

```diff
@@ -55,54 +55,26 @@ public static int Start(string consoleFilePath, [MarshalAs(UnmanagedType.LPArray
             Thread.CurrentThread.CurrentUICulture = NativeCultureResolver.UICulture;
             Thread.CurrentThread.CurrentCulture = NativeCultureResolver.Culture;
 
-            RunspaceConfigForSingleShell configuration = null;
-            PSConsoleLoadException warning = null;
             //      PSSnapInException will cause the control to return back to the native code
             //      and stuff the EXCEPINFO field with the message of the exception.
             //      The native code will print this out and exit the process.
-            if (string.IsNullOrEmpty(consoleFilePath))
-            {
 #if DEBUG
-// Special switches for debug mode to allow self-hosting on InitialSessionState instead
-// of runspace configuration...
-                    if (args.Length > 0 && !String.IsNullOrEmpty(args[0]) && args[0].Equals("-iss", StringComparison.OrdinalIgnoreCase))
-                    {
-                        ConsoleHost.DefaultInitialSessionState = InitialSessionState.CreateDefault2();
-                        configuration = null;
-                    }
-                    else if (args.Length > 0 && !String.IsNullOrEmpty(args[0]) && args[0].Equals("-isswait", StringComparison.OrdinalIgnoreCase))
-                    {
-                        Console.WriteLine("Attach the debugger and hit enter to continue:");
-                        Console.ReadLine();
-                        ConsoleHost.DefaultInitialSessionState = InitialSessionState.CreateDefault2();
-                        configuration = null;
-                    }
-                    else
-                    {
-                        ConsoleHost.DefaultInitialSessionState = InitialSessionState.CreateDefault2();
-                        configuration = null;
-                    }
-#else
-                ConsoleHost.DefaultInitialSessionState = InitialSessionState.CreateDefault2();
-                configuration = null;
-#endif
-            }
-            else
+            // Special switch for debug mode
+            if (args.Length > 0 && !String.IsNullOrEmpty(args[0]) && args[0].Equals("-isswait", StringComparison.OrdinalIgnoreCase))
             {
```

Comment:
I'd move this code earlier, like in main. I needed similar code in the past and never noticed it here.
My version was little nicer, waiting for the debugger to attach (in a loop, with time outs) and then hit a breakpoint.

Reply:
Is this code needed anymore?  In VSCode, it will break in main() anyways.  Plus this code looks to be an artifact of when ISS was new as it's blocking at the creation of ISS.

Reply:
Sometimes you need to launch from another process, so it can be useful. But you could just remove it and the next time it's needed, something like it will come back.

Reply:
Will remove